Plugins~/Src/MeshSyncClientBlender/python/unity_mesh_sync_installation.py

```diff
@@ -0,0 +1,204 @@
+
+def msb_get_editor_path_suffix():
+    os = platform.system()
+    if os == 'Windows':
+        path = "\\Editor\\Unity.exe"
+    elif os == 'Darwin':
+        path = "/Unity.app/Contents/MacOS/Unity"
+    elif os == 'Linux':
+        path = "/Unity.app/Contents/Linux/Unity"
+    return path
+
+def msb_get_editor_path(editor_version):
+    return msb_get_editor_path_prefix() + editor_version + msb_get_editor_path_suffix()
+
+def msb_validate_project_path(directory):
+    project_version_path = directory +"/ProjectSettings/ProjectVersion.txt"
+    if path.exists(project_version_path):
+        return True
+    return False
+
+def msb_try_get_valid_project_path(context):
+    directory = context.scene.meshsync_unity_project_path
+
+    #Validate that the path is a project
+    if msb_validate_project_path(directory) == False:
+        #Try to fill a valid path from an already running project
+        directory = msb_try_get_path_from_server()
+        if msb_validate_project_path(directory) == False:
+            return 'INVALID_PATH'
+
+    return directory
+
+def msb_try_get_path_from_server():
+    if not msb_context.is_editor_server_available:
+        return 'NO SERVER'
+
+        #Get the project path
+    msb_context.sendEditorCommand(EDITOR_COMMAND_GET_PROJECT_PATH)
+    server_reply = msb_context.editor_command_reply
+    return server_reply
+
+def msb_add_meshsync_to_unity_manifest(manifest_path, lock_path, entry):
+
+    with open(lock_path, "r+") as file:
+        data = json.load(file);
+    
+        # check if MeshSync is installed on selected project
+        # if not installed, install it
+        found = False
+
+        for package in data['dependencies']:
+            if package == 'com.unity.meshsync' and data['dependencies']['com.unity.meshsync']['version'] == entry:
+                found = True
+                break
+            elif package == 'com.unity.meshsync':
+                print(data['dependencies']['com.unity.meshsync'])
+
+    if found == False:
+        with open(manifest_path, "r+") as file:
+            data = json.load(file);
+            #install for user
+            dependencies = data["dependencies"];
+            dependencies["com.unity.meshsync"] = entry
+            file.seek(0)
+            file.truncate(0)
+            json.dump(data, file)
+
+def msb_try_install_meshsync_to_unity_project(directory):
+
+        if directory[-1] == '/' or directory[-1] == '\\':
+            lock_path = directory + "Packages/packages-lock.json"
+            manifest_path = directory + "Packages/manifest.json"
+        else:
+            lock_path = directory + "/Packages/packages-lock.json"
+            manifest_path = directory + "/Packages/manifest.json"
+
+        #TODO replace with release that has the Editor Commands changes
+        manifest_entry = "0.14.5-preview"
+        if DEBUG == 1:
+            manifest_entry = "file:C:\\Users\\Sean Dillon\\MeshSync\\MeshSync~\\Packages\\com.unity.meshsync"
```

Comment:
Probably meant to delete this debug code?

Reply:
This is necessary for testing. I will keep this comment unresolved until we make the change

Reply:
Removed the code